timemachine/maps/terminal_bonds.py

```diff
@@ -0,0 +1,239 @@
+"""Summarize HarmonicBonds as Intervals in R+, construct maps between Intervals, & apply these maps to bond lengths"""
+
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+from jax import config, jacobian, jit
+from jax import numpy as jnp
+from jax import vmap
+from numpy.typing import NDArray as Array
+
+config.update("jax_enable_x64", True)
+
+from typing import List
+
+import networkx as nx
+
+from timemachine.constants import BOLTZ, DEFAULT_TEMP
+
+
+@dataclass
+class Interval:
+    lower: float
+    upper: float
+
+    @property
+    def width(self) -> float:
+        return self.upper - self.lower
+
+    def validate(self) -> None:
+        assert self.width > 0
+        assert self.lower > 0
+
+
+@dataclass
+class Gaussian:
+    mean: float
+    stddev: float
+
+    def to_interval(self, sigma_thresh=20) -> Interval:
+        r = self.stddev * sigma_thresh
+        interval = Interval(self.mean - r, self.mean + r)
+        interval.validate()
+        return interval
+
+
+def gaussians_from_harmonic_bonds(ks, eq_lengths, temperature=DEFAULT_TEMP) -> List[Gaussian]:
+    kBT = BOLTZ * temperature
+    params = zip(ks, eq_lengths)
+    return [Gaussian(eq_length, np.sqrt(kBT / k)) for (k, eq_length) in params]
+
+
+@jit
+def interval_map(x, src_lb, src_ub, dst_lb, dst_ub):
+    """map x in Interval[src_lb, src_ub] to y in Interval[dst_lb, dst_ub]"""
+
+    scale_factor = (dst_ub - dst_lb) / (src_ub - src_lb)
+
+    in_support = (x >= src_lb) * (x <= src_ub)
```

Comment:
nit: should we prefer `&` for logical AND?

Reply:
Yes, that would be clearer!

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/801/commits/0f07fc1cbefb6cf07377d7dd130aa724b861e84d